File: src/main/java/com/cfgglobal/generator/task/processor/MultipleTaskProcessor.java
package com.cfgglobal.generator.task.processor;

import com.cfgglobal.generator.entity.CodeEntity;
import com.cfgglobal.generator.entity.CodeProject;
import com.cfgglobal.generator.entity.Task;
import com.cfgglobal.generator.task.service.TaskService;
import com.google.common.collect.Lists;

import java.util.List;
import java.util.Map;

public class MultipleTaskProcessor implements ITaskProcessor {
    @Override
    public List<String> run(CodeProject codeProject, Task task, Map<String, Object> context) {
        List<String> paths = Lists.newArrayList();
        for (CodeEntity codeEntity : codeProject.getEntities()) {
            context.put("entity", codeEntity);
            codeProject.getTemplateEngine().put("entity", codeEntity);
            paths.add(TaskService.processTemplate(codeProject, task, context));
        }
        return paths;

    }
}

File: src/main/kotlin/com/cfgglobal/generator/task/processor/MultipleTaskProcessor.kt
package com.cfgglobal.generator.task.processor

import com.cfgglobal.generator.entity.CodeProject
import com.cfgglobal.generator.entity.Task
import com.cfgglobal.generator.task.service.TaskService
import com.google.common.collect.Lists

class MultipleTaskProcessor : ITaskProcessor {
    override fun run(codeProject: CodeProject, task: Task, context: MutableMap<String, Any>): List<String> {
        val paths = Lists.newArrayList<String>()
        for (codeEntity in codeProject.entities) {
            context["entity"] = codeEntity
            codeProject.templateEngine.put("entity", codeEntity)
            paths.add(TaskService.processTemplate(codeProject, task, context))
        }
        return paths

    }
}
